mk/Common.mk

```diff
@@ -505,6 +505,7 @@ else
 		-nostartfiles \
 		$(TEST_FILES) \
 		-T $(LD_FILE) \
+		-Xlinker -Map=$*.map \
```

Comment:
`-Xlinker` passes the rest of this line to `ld`?

`-Map=$*.map` causes a new `mytest.elf.map` file to be written next to the elf?

Reply:
This gcc option passes the next option/token to the linker. This is used to pass options down to the linker that gcc does not understand. https://gcc.gnu.org/onlinedocs/gcc/Link-Options.html#index-Xlinker

The `$*` automatic variable will match the stem of the target which is the full path before `.elf` in our case. If for instance the test output/target is called `mytest.elf` then the map file will be called `mytest.map`. This also means that the map file will be placed next to the elf file like the other metadata files we generate. https://www.gnu.org/software/make/manual/html_node/Automatic-Variables.html

Reply:
Perfect